server.go

```diff
@@ -0,0 +1,79 @@
+package baseplate
+
+import (
+	"errors"
+	"io"
+	"os"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/secrets"
+)
+
+// ServerConfig is a general purpose config for assembling a BaseplateServer
+type ServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Log struct {
+		Level log.Level
+	}
+
+	Metrics struct {
+		Namespace           string
+		Endpoint            string
+		CounterSampleRate   float64
+		HistogramSampleRate float64
```

Comment:
🔕 how does yaml handle default values? if we change the 2 sample rates into `*float64`, will they be `nil` when that part is absent from the config file? if that's the case we would want to use pointers for the sample rates.